### PR TITLE
adding hydra configuration

### PR DESCRIPTION
**Pull request by Loic-Ker**

### CODE REVIEW

**Comment by peterampazzo** [comment]
Not sure if keeping configurations in `src/models/conf` is the best location.

**Comment by peterampazzo** [comment]
Also, the PR should be merged to `develop` and not `code-style` 😉

File: src/models/train_model.py [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+import logging
 
+log = logging.getLogger(__name__)
 
-def train(epochs=20, lr=0.001):
+@hydra.main(config_path="config", config_name='default.yaml')
+
+def train(config):
     print("Training")
+    print(f"configuration: \n {OmegaConf.to_yaml(config)}")
```
Comment (erlahrafnkels) [inline src/models/train_model.py]: @peterampazzo had a comment about changing "print" to "log" in lines 13 to 14, right? Should it also be done in line 44? @peterampazzo

**Review by erlahrafnkels — APPROVED** [review]

File: src/models/train_model.py [hunk 2]
```diff
@@ -20,21 +31,20 @@ def train(epochs=20, lr=0.001):
...
             optimizer.step()
-
+        
             running_loss += loss.item()
 
         print(f"Epoch: {e} - Training loss: {running_loss/len(train_set):5f}")
         train_loss.append(running_loss / len(train_set))
-
-    torch.save(model.state_dict(), "models/trained_model.pt")
+    torch.save(model, "models/trained_model.pt")
```
Comment (alessandro-canevaro) [inline src/models/train_model.py]: Isn't it better to save `model.state_dict()`?
(https://stackoverflow.com/questions/42703500/best-way-to-save-a-trained-model-in-pytorch)

**Pull request merged**